### ISSUE TITLE
Jerky Movement of ViewAnnotation

### ISSUE THREAD

**Issue by swasta** [mapbox/mapbox-maps-ios#2163]
## Environment

- Xcode version: 15.2
- iOS version: 17.2
- Devices affected: all
- Maps SDK Version: 11.3.0

## Observed behavior and steps to reproduce

When adding a `ViewAnnotation` to the map and updating its coordinates to the camera center upon camera changes, the annotation moves jerkily, especially noticeable when dragging the map quickly. This issue occurs in Maps SDK Version 11.3.0.

1. Add a `ViewAnnotation` on the map with an initial coordinate.
2. Subscribe to the `cameraChanged` event.
3. In the `cameraChanged` event, update the annotation's coordinate to the camera center.
4. Drag the map quickly.
5. Observe that the annotation's movement is jerky.

```swift
let annotation = ViewAnnotation(
    coordinate: .init(latitude: 40.7128, longitude: -74.0060),
    view: SimplePointerView()
)

mapView.viewAnnotations.add(annotation)

mapView.mapboxMap.onCameraChanged
    .observe { cameraChange in
        annotation.annotatedFeature.geometry = .point(.init(cameraChange.cameraState.center))
    }
    .store(in: &cancellables)
```

## Expected behavior

The annotation is expected to move smoothly across the map in response to camera changes, similar to the behavior observed in Maps SDK Version 10.17.0.

## Notes / preliminary analysis

In our actual project, we're not just sticking the annotation's view to the camera's center. I'm doing this in the issue to demonstrate the problem in the simplest way. Actually, we're using our own calculated coordinates to update the annotations' positions and observing this stuttering movement. With this upgrade from v10 to v11 throwing us this curveball, we're kind of thinking about ditching moving annotations through Mapbox and maybe just sticking UIViews on top of the mapView instead. Not exactly thrilled about having to possibly go this route because of the update. It’s extra work we weren’t planning on.

## Additional links and references

Video demonstrating jerky movement in v11.3.0: [Link to video](https://github.com/mapbox/mapbox-maps-ios/assets/5530878/35872425-2d9a-4dbe-81d8-c9e2f6971c99)

Video showing smooth movement in v10.17.0: [Link to video](https://github.com/mapbox/mapbox-maps-ios/assets/5530878/b02d5af4-f57e-497e-b380-0f9bab9a42c8)

**Comment by persidskiy** [comment]
Hi @swasta , can you try the setting the `presentsWithTransaction = true`? Usually, it helps with problems with annotations

https://docs.mapbox.com/ios/maps/api/11.3.0/documentation/mapboxmaps/mapview/presentswithtransaction

**Comment by swasta** [comment]
> can you try the setting the presentsWithTransaction = true? Usually, it helps with view annotations issues

Unfortunately, setting this property doesn't seem to change anything

**Comment by swasta** [comment]
@persidskiy  Yes, that’s exactly right! Imagine we have one ViewAnnotation positioned in the middle of the screen. When the user drags the map causing the annotation to move outside the visible viewport, we want to maintain a pointer—a UIView—close to the edge of the screen where the annotation exited. This visual cue informs users that they’ve panned the annotation off-screen and indicates which direction they should drag the map to bring the annotation back into view. To accomplish this, we calculate the pointer's coordinate so it’s adjacent to the appropriate screen boundary. With each movement of the camera, we recalculate this coordinate and adjust the pointer's position accordingly. This mechanism worked flawlessly in SDK version 10.17.0, but we've noticed that the movement has lost its smoothness after upgrading to version 11.3.0.

> Do you need to use view annotations then?

You're probably right; we might not necessarily need to use them. However, as mentioned, this functionality was working perfectly until we decided to upgrade to version 11.
